File: cpp/binarytodecimal.cpp
#include<iostream>
#include "stack.cpp"
using namespace std;
class binary{
	int num,bit,del;
	stack<int>st;
	public:
		void read();
		void display();};

void binary::read(){
	cout<<"enter the decimal number\n";
	cin>>num;
	dec=num;
	while(num>0){
		bit=num%2;
		s1.push(bit);
		num=num/2;}}

void binary::displaty(){
	cout<<"\n the binary of the vdecimal number"<<dec<<"is:";
	while(!stactempty()){
		cout<<st.pop();}
	cout<<"\n";}

int main(){
	binary b;
	b.read();
	b.display();
	return 0;}

File: python/binarytodecimal.py
def bin_to_dec(x):
    l=list(x)
    l=list(map(int,l))
    bin=j=0
    for i in range(len(l)-1,-1,-1):
        bin+=l[i]*2**j
        j+=1
    return bin
x=input("Enter a Binary Number: ")
print(bin_to_dec(x))